### ISSUE TITLE
Call to function ‘ExpoLocation.startLocationUpdatesAsync’ has been rejected. → Caused by: Couldn’t start the foreground service. Foreground service cannot be started when the application is in the background

### ISSUE THREAD

**Issue by geikobogdan** [expo/expo#32545]
### Summary

Location tracking doesn't work in background with foreground service on Android. On IOS it works fine:
<img width="416" alt="Screenshot 2024-11-01 at 19 15 55" src="https://github.com/user-attachments/assets/69f230f5-8516-415c-8fbc-63561bc9d4df">

my app config:
`    "android": {
        "permissions": [
          "RECEIVE_BOOT_COMPLETED",
          "WAKE_LOCK",
          "ACTION_BOOT_COMPLETED",
          "ACCESS_FINE_LOCATION",
          "ACCESS_COARSE_LOCATION",
          "ACCESS_BACKGROUND_LOCATION",
          "SCHEDULE_EXACT_ALARM",
          "FOREGROUND_SERVICE"
        ],
    },
"plugins": [
      [
        "expo-location",
        {
          "locationAlwaysAndWhenInUsePermission": "Allow $(PRODUCT_NAME) to use your location to connect you with client.",
          "isIosBackgroundLocationEnabled": true,
          "isAndroidForegroundServiceEnabled": true,
          "isAndroidBackgroundLocationEnabled": true
        }
      ],]
`

Task Script:

```
await Location.startLocationUpdatesAsync(LOCATION_TASK_NAME, {
            accuracy: Location.Accuracy.Balanced,
            activityType: Location.ActivityType.AutomotiveNavigation,
            distanceInterval: 10,
            showsBackgroundLocationIndicator: true,
            foregroundService: {
              notificationTitle: 'You are on duty',
              notificationBody:
                'Operator app is tracking your location (only when you are on duty)',
              // notificationColor: '#2C63B6',
            },
            pausesUpdatesAutomatically: false,
          })
```

### Managed or bare workflow?

managed

### What platform(s) does this occur on?

Android

### Package versions

"expo": "~51.0.38",
"expo-location": "~17.0.1",
"expo-task-manager": "11.8.2",

### Environment

```text
dev
```

### Reproducible demo

default expo react native app

**Comment by brentvatne** [comment]
hey there, i recommend trying out another library if this isn't working well for you right now. we try to warn about this in the docs: https://docs.expo.dev/versions/latest/sdk/location/#background-location-methods

we are currently working on improving our background tasks handling overall, and background location will be improved as part of that.

**Comment by geikobogdan** [comment]
Hey @brentvatne, thx a lot for a quick response. Unfortunately, my team is limited in time and we have a lot of logic depends of `expo-location` and `startLocationUpdatesAsync`. Also this library you recommend is paid for android :(
Could you please clarify when this feature will be stable (approximate ETA)?

**Comment by brentvatne** [comment]
@geikobogdan - we don't have a timeline for that, but @chrfalch is working on improving background tasks at the moment. you might be able to find an alternative option here: https://reactnative.directory/?search=location

**Comment by geikobogdan** [comment]
@brentvatne yeah, I see, but as I mentioned before we already use `expo-location` in production and replace it with another solution can take some time (testing, .etc). Okay then I'll try to use another library. Hope it will be resolved soon!

**Comment by geikobogdan** [comment]
I suggest closing this issue only after the problem is fixed in future versions. Thanks for understanding!

**Comment by cs2812** [comment]
@geikobogdan track location in the background is working fine. I tried it with the new expo version 52. it is working in all app modes foreground, background, and killed.

**Comment by chrfalch** [comment]
I'm working my way through `expo-location` and was wondering if we could try to narrow down what's going on here. @geikobogdan - could you provide a super small reproduction of the problem that I could use for testing?

**Comment by cs2812** [comment]
just create a new expo app and follow this doc - https://docs.expo.dev/versions/latest/sdk/task-manager/

**Comment by chrfalch** [comment]
This doesn't reproduce the error reported by the original poster (@geikobogdan) - and as you say it works as expected.

**Comment by lrosasarias05** [comment]
@puelocesar did you find any solution to this? I'm facing the exact same issue, but only for the "preview build" (compiled with EAS). I don't have any problem when working with the development build.

**Comment by chrfalch** [comment]
Please create a small repro that displays the problem. This will help us immensely since it shows how you want to use the API and the errors - without it is impossible to debug the issue further. We have test apps that use the API as it is documented (or we believe they are) and background locations are working well on Android in these tests.

**Comment by Rodrigo816** [comment]
###  `startLocationUpdatesAsync` Can't Be Used with App State on Android (Expo SDK 52)

I'm experiencing an issue on **Expo SDK 52** where `startLocationUpdatesAsync` **cannot be started when the app transitions to the background** on **Android**. This works fine on **iOS**, but on Android, I can only start `startLocationUpdatesAsync` while the app is open.

This limitation creates a problem because:  
1. **App State-Based Logic No Longer Works** – Previously, I used `AppState` to determine when the app goes to the background and start the background location task. However, since Android now blocks starting a foreground service from the background, this logic fails.
2. **Two Listeners Running Simultaneously** – To work around this, I would need to always start `startLocationUpdatesAsync` when the app is open, but this means I end up with:  
   - `startLocationUpdatesAsync` running as the background location tracker.  
   - `watchPositionAsync` running for real-time updates in the foreground.  
   - ❌ **This is not optimal and causes a lot of issues.**  

---

## **Expected Behavior**
- `startLocationUpdatesAsync` should be allowed to start **when the app transitions to the background**.  
- The app should not be required to **always start the background task while it's open**, just to ensure it runs in the background later.  
- We should not need to run **two location listeners simultaneously** to work around this issue.

---

## **Actual Behavior**
- On **iOS**, `startLocationUpdatesAsync` can be started in the background just fine.  
- On **Android (SDK 52)**, `startLocationUpdatesAsync` **fails when the app state changes to background**.  
- This forces me to start it **while the app is in the foreground**, leading to **two active listeners** at the same time, which is inefficient and problematic.  


Related Issues:
https://github.com/expo/expo/issues/28959

**Comment by Rodrigo816** [comment]
Seams like if I just call startLocationUpdatesAsync works in foreground and background so I don't need watchPositionAsync, I was able to use defineTask inside a screen to update in real time. Expo's documentation recommends defining defineTask in App.tsx or the root of your project.

```
  TaskManager.defineTask(LOCATION_TASK_NAME, async ({ data, error }) => {
    if (error) {
      console.error("Background Task Error:", error);
      return;
    }

    if (data) {
      const { locations } = data;
      if (locations.length > 0) {
        console.log("📍 Background Location Update:", locations[0]);
        setCurrentLocation(locations[0]); 
      }
    }
  });
```

**Comment by FarruxNosirov** [comment]
Error: Call to function 'ExpoLocation.startLocationUpdatesAsync' has been rejected.
→ Caused by: Couldn't start the foreground service. Foreground service permissions were not found in the manifest

bu muammoni kimdur hal qila oladimi ?

**Comment by FarruxNosirov** [comment]
"plugins": [
...,
    [
        "expo-location",
        {
          "locationAlwaysAndWhenInUsePermission": "Allow $(PRODUCT_NAME) to use your location.",
          "locationAlwaysPermission": "Allow $(PRODUCT_NAME) to use your location",
          "locationWhenInUsePermission": "Allow $(PRODUCT_NAME) to use your location",
          "isIosBackgroundLocationEnabled": true,
          "isAndroidBackgroundLocationEnabled": true,
          "isAndroidForegroundServiceEnabled": true
        }
      ],
   [
        "expo-build-properties",
        {
          "android": {
            "enableProguardInReleaseBuilds": false
          }
        }
      ],
],
   "android": {
..
      "permissions": [
        "android.permission.CAMERA",
        "android.permission.READ_EXTERNAL_STORAGE",
        "android.permission.WRITE_EXTERNAL_STORAGE",
        "android.permission.ACCESS_FINE_LOCATION",
        "android.permission.ACCESS_COARSE_LOCATION",
        "android.permission.ACCESS_BACKGROUND_LOCATION",
        "android.permission.FOREGROUND_SERVICE",
        "android.permission.POST_NOTIFICATIONS",
        "android.permission.RECORD_AUDIO",
        "android.permission.FOREGROUND_SERVICE_LOCATION",
        "android.permission.FOREGROUND_SERVICE_TYPE_LOCATION",
        "android.permission.FOREGROUND_SERVICE_CAMERA",
        "android.permission.FOREGROUND_SERVICE_MICROPHONE",
        "android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK"
      ],
  "foregroundServiceType": ["location"],
}
I solved this problem this way.
this worked on android 14 version without any problems, in the version "expo": "~52.0.35"
